segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/AsyncBaseChunkStorage.java

```diff
@@ -661,17 +682,32 @@ protected void checkChunkName(String chunkName) {
      * StreamSegmentExceptions.
      *
      * @param operation The function to execute.
+     * @param opContext Context for the given operation.
      * @param <R>       Return type of the operation.
      * @return CompletableFuture<R> of the return type of the operation.
      */
-    protected <R> CompletableFuture<R> execute(Callable<R> operation) {
+    protected <R> CompletableFuture<R> execute(Callable<R> operation, OperationContext opContext) {
         return CompletableFuture.supplyAsync(() -> {
             Exceptions.checkNotClosed(this.closed.get(), this);
             try {
-                return operation.call();
+                val timer = new Timer();
+                val ret = operation.call();
+                opContext.setInclusiveLatency(timer.getElapsed());
+                return ret;
             } catch (Exception e) {
                 throw new CompletionException(e);
             }
         }, executor);
     }
+
+    /**
+     * Context data used to capture operation runtime information.
+     */
+    @Data
```

Comment:
`@Data` will also create  a constructor. Since you never assign this via the constructor, change this to `@Getter` + `@Setter`.

Reply:
Fixed